firmware/controllers/can/can_dash_ms.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file	can_dash_ms.cpp
+ *
+ * This file implements MegaSquirt CAN realtime data broadcasting format
+ *
+ * @date May 8, 2023
+ * @author AndreyGusakov, (c) 2023
+ */
+
+#include "pch.h"
+
+#if EFI_CAN_SUPPORT
+#include "can.h"
+#include "can_msg_tx.h"
+
+#include "rusefi_types.h"
+#include "rtc_helper.h"
+#include "fuel_math.h"
+
+/* TODO:
+ * use engine->outputChannels instead of Sensor::getOrZero as it cheaper */
```

Comment:
no, no-no-no!!! Definitely not read from engine->outputChannels

Reply:
Don't we care about coherency and cpu time? Why do same job twice?

Reply:
it's a balance between design risks and costs. the code here is not on the mission critical path so we can totally afford x5 CPU consumption for 20% readability improvement.

humans are expensive, CPU is cheap

Reply:
Using `outputChannels` is not cheaper at all. If TS isn't connected, there shouldn't be a requirement to populate it at all! There is no guarantee that it is populated, in fact.